### PR TITLE
Move autoscaling e2e tests to a separate directory

### PR DESCRIPTION
**Pull request by mwielgus**
For fine-grain access control. Autoscaling team is expanding the e2e test coverage and the need for getting an approval for every PR is annoying. 

cc: @MaciekPytel @jszczepkowski @fgrzadkowski @wojtek-t

### CODE REVIEW

File: test/e2e/autoscaling/OWNERS [hunk 1]
```diff
@@ -0,0 +1,8 @@
+reviewers:
```
Comment (wojtek-t) [inline test/e2e/autoscaling/OWNERS]: I don't have place to comment, so commenting here.
This won't work out of the box - you need to also modify:
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/e2e_test.go#L25

File: test/e2e/autoscaling/cluster_size_autoscaling.go [hunk 2]
```diff
@@ -36,6 +36,7 @@ import (
 	policy "k8s.io/client-go/pkg/apis/policy/v1beta1"
 	"k8s.io/kubernetes/pkg/api/v1"
 	"k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+	. "k8s.io/kubernetes/test/e2e"
```
Comment (wojtek-t) [inline test/e2e/autoscaling/cluster_size_autoscaling.go]: I'm not a fan of it. How many places you would need to change to make it work?

File: test/e2e/kubelet.go [hunk 3]
```diff
@@ -98,7 +98,7 @@ func waitTillNPodsRunningOnNodes(c clientset.Interface, nodeNames sets.String, p
 // In case a given label already exists, it overwrites it. If label to remove doesn't exist
 // it silently ignores it.
 // TODO: migrate to use framework.AddOrUpdateLabelOnNode/framework.RemoveLabelOffNode
-func updateNodeLabels(c clientset.Interface, nodeNames sets.String, toAdd, toRemove map[string]string) {
+func UpdateNodeLabels(c clientset.Interface, nodeNames sets.String, toAdd, toRemove map[string]string) {
```
Comment (wojtek-t) [inline test/e2e/kubelet.go]: Can you please move this function to framework? I don't want to depend on test/e2e .

File: test/e2e/autoscaling/dns_autoscaling.go [hunk 4]
```diff
@@ -28,6 +28,7 @@ import (
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/kubernetes/pkg/api/v1"
 	"k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+	. "k8s.io/kubernetes/test/e2e"
```
Comment (wojtek-t) [inline test/e2e/autoscaling/dns_autoscaling.go]: Same here

File: test/e2e/kubelet.go (hunk 3 above)
Comment (mwielgus) [inline test/e2e/kubelet.go]: not relevant anymore.

File: test/e2e/autoscaling/dns_autoscaling.go (hunk 4 above)
Comment (mwielgus) [inline test/e2e/autoscaling/dns_autoscaling.go]: done.

File: test/e2e/autoscaling/cluster_size_autoscaling.go (hunk 2 above)
Comment (mwielgus) [inline test/e2e/autoscaling/cluster_size_autoscaling.go]: Done.

File: test/e2e/autoscaling/OWNERS (hunk 1 above)
Comment (mwielgus) [inline test/e2e/autoscaling/OWNERS]: Done.

**Comment by wojtek-t** [comment]
Seems like a import cycle problem.

File: test/e2e/framework/size.go [hunk 5]
```diff
@@ -0,0 +1,128 @@
+/*
```
Comment (wojtek-t) [inline test/e2e/framework/size.go]: I don't like the "size.go" name, but I don't have better suggestion, so let's leave :)

**Comment by mwielgus** [comment]
@k8s-bot non-cri e2e test this

**Pull request merged**